Make JNIEnv `Copy`

jack-signal opened the pull request:
## Overview

Currently JNIEnv is `Clone` but not `Copy`, however there seems no difficulty in having it also be `Copy`.

Possibly I've missed or misunderstood something such that this is a bad idea, in which case I look forward to learning the reason. Thank you.

If this is acceptable but needs some additional test or documentation work just let me know.

### Definition of Done

- [x] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/jni-rs/jni-rs/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/jni-rs/jni-rs/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has documentation
- [x] This change is not breaking **or** mentioned in the Changelog
- [ ] The [continuous integration build](https://www.travis-ci.org/jni-rs/jni-rs) passes

stanislav-tkach reviewed (approved):
The [official documentation](https://doc.rust-lang.org/std/marker/trait.Copy.html#when-should-my-type-be-copy) suggests to implement `Copy` for any type that implements `Clone` with some forward compatibility consideration. JNIEnv is cheap to copy, so I guess it is a good idea.

stanislav-tkach commented:
@jack-signal Thanks for the pull request. Could you please update the changelog?

jack-signal commented:
I updated the changelog and rebased against master to avoid a merge conflict. I added the note under `Changes` subheading as that is where similar trait additions were placed (eg `Implemented Sync for GlobalRef`)

dmitry-timofeev commented:
Thanks @jack-signal , merged!
